### ISSUE TITLE
Inconsistent Results

### ISSUE THREAD

**Issue by AndrewJohnson150** [fkie-cad/cwe_checker#220]
Hello, while running cwe_checker I noticed that I seem to get different results with each run. I'm unsure of what the cause could be. 

I've included an example of what is happening. Below you can see two calls to run cwe_checker on the same file, but I receive different results both times.

Is this intended behavior? What could be causing this?

```
PS C:...> docker run --rm -v C:\...busybox-1.30.1_lite_busybox_unstripped_x86_64:/input fkiecad/cwe_checker:latest --quiet /input
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ad4 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00405762 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00405888 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0041824a may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040f1ee may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b509 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004082a8 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00404466 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0041afea may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00448201 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004271dd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040cdbb may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440acd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004270dc may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004058ae may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b4c9 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040e875 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0044cd50 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b461 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004138b7 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042d305 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004147e8 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f094 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00408827 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004311a6 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004058db may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00427beb may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004087f9 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00405836 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004047da may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004182b3 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00427bfa may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040e8a1 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00408292 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408820 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004059db may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00445940 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408845 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ac6 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00403cc0 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00404472 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00416bb2 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b4b5 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f105 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004155d5 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0044594c may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408852 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f16b may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b487 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042e9c8 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ac0 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00417399 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ae2 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00405904 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004176a3 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440adb may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040f113 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004058bd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042ae3c may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040eb77 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042d277 may be out of bounds


PS C:\...> docker run --rm -v C:\...busybox-1.30.1_lite_busybox_unstripped_x86_64:/input fkiecad/cwe_checker:latest --quiet /input
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00404472 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00448201 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f105 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00408827 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040f1ee may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b4b5 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408852 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00408292 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00417399 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ac0 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004176a3 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ac6 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042d277 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00427bfa may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408845 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042ae3c may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004147e8 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040e875 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004138b7 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ad4 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f16b may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00403cc0 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0044f094 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042d305 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004182b3 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004311a6 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b509 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440acd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040cdbb may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00408820 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0044594c may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0040e8a1 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004087f9 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040eb77 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0040f113 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b4bd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440adb may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004047da may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b4c9 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004271dd may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004082a8 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042e9c8 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0044cd50 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 004270dc may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00440ae2 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00416bb2 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 0041824a may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b487 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00404466 may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 00427beb may be out of bounds
[CWE125] (0.2) (Out-of-bounds Read) Memory load at 0042b461 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 004155d5 may be out of bounds
[CWE787] (0.2) (Out-of-bounds Write) Memory write at 00445940 may be out of bounds
```

**Comment by Enkelmann** [comment]
No, this is not intended behavior. In your case it seems that the results are just ordered differently (i.e. more or less randomly) in both runs. I consider this a bug and will look into it.

In some very rare cases I also observed nondeterministic disassembly by Ghidra, which could in theory also cause nondeterministic cwe_checker output. But I do not think that this affects your binary.

**Comment by Enkelmann** [comment]
The issue should be fixed now on the master branch thanks to PR #221. However, currently the automated docker image builds on Dockerhub are not working. We are working on it but it may take some days until the automated builds work again. Until then the fix won't be included in the `fkiecad/cwe_checker:latest` docker image.
